### DISCUSSION TITLE
RaspberryMatic als HA Addon - Zugriff TinyMatic oder anderer Drittanbieter-Apps

### DISCUSSION THREAD

**Discussion by Columbus1964** [OpenCCU/OpenCCU#1774] [🇩🇪 Fragen&Antworten] [answered]
Hallo, ich nutze HA in einer VM auf einer Synology-NAS. Im HA habe ich das Raspberrymatic-Addon von Jens Maus installiert und meine bisherige, auf einen RPi laufende Raspberrymatic in das Addon umgezogen und angepasst (z.B. Homematic-Zusatzsoftware). Für den Funk verwende ich einen USB-Stick RFUSB. Klappt auch alles soweit prima, besten Dank für die geleistet Arbeit. Nun habe ich aber Apps (noch AIO-Neo, Tinymatic; oder auch JSmatic), die mit der Homematic per IP:Port kommunizieren. Das Addon hat ja aber keine eigene IP. In der YAML vom HA steht für Host statt der IP jetzt die Kennung des Addons. Greift man auf die Oberfläche der Raspberrymatic über die Seitenleiste im HA zu, steht im Browser http://ip-des-HA:8123/kennung-des-addons/dashboard. Mir ist es nicht gelungen in jeglicher Kombination der Adresse mit den oben genannten Apps auf die Raspberrymatic zuzugreifen. Ist das bei Verwendung des Addon nicht möglich? Wäre es also besser, die RaspberryMatic in einer 2. VM mit eigener IP laufen zu lassen? Ich dachte, dass die Integration in HA als Addon vielleicht tiefer (besser) ist, als in einer separaten VM oder wie bisher auf einen RPi. Kann mir hierzu jemand helfen?
VG Columbus1964

**Comment by Baxxy13** [comment, marked as answer]
Also mit Tinymatic auf das RM-HA-AddOn zuzugreifen ist eine der leichteren Übungen.
Tinymatic nutzt das XML-API AddOn und somit muss nur der Standard WebUI - Port verfügbar sein.

- innerhalb des RM-HA-AddOns (also in der WebUI von RaspberryMatic) die Firewall zunächst komplett öffnen (hier kann man später wieder dichtmachen wenn man will, nötig ist das nicht)
- in der Homeassistant - UI beim RM-HA-AddOn zur Konfiguration gehen und dort den Port 80 "nach außen" auf einen Port deiner Wahl freigeben. (Da Homeassistant selbst den Port:80 nicht nutzt kann man den sogar nehmen)
Beispiel:

![RM-HA_AddOn_Ports_nach_aussen](https://user-images.githubusercontent.com/54898249/160397867-ee1e19e5-b570-4f98-93b8-7960c29efcaf.JPG)

- in Tinymatic bei Serveradresse die IP deines Homeassistant-Hosts mit angehängtem Port eingeben

Beispiel:
192.168.1.66:8088

Fertig!

Das ist das Grundprinzip. 
Für die NEO-Remote z.B. müsste man also erstmal beim Anbieter erfragen auf welche Ports der Zentrale der Zugriff erfolgt.
Ist es bspw: 8181 dann leitet man den eben 1:1 nach außen weiter.

**Reply by Columbus1964** [reply]
Hallo Baxxy13,

recht herzlichen Dank für die Unterstützung und die gute Erklärung. Jetzt habe ich das Grundprinzip der Portangaben in den Konfigurationen von HA-Addons verstanden und das nützt mir nicht nur in Bezug auf diese Anfrage, sondern generell. TinyMatic funktioniert und auch der Zugriff per Browser ist so angenehmer als über den HA.

**Comment by fastbag** [comment]
Funktioniert alles nicht
